Chris3 (#379)
LiskHQ/lisk-docs: modules/ROOT/pages/configuration.adoc

To set up a public wallet, simply leave the `modules.http_api.access.whitelist` array empty.

For optimal security, disable all access setting `modules.http_api.enabled` to `false`.

NOTE: This last configuration may prevent monitoring scripts from functioning.

== Forging

To enable your node to forge, firstly it is required to insert some encrypted data into the config file under the `chain.forging.delegates` array.
To encrypt your passphrase, it is recommended to use one of the following alternatives listed below:

* xref:{v_sdk}@lisk-sdk::lisk-commander/user-guide/commands.adoc[Lisk Commander] via `encrypt passphrase` command.

> * xref:{v_sdk}@lisk-sdk::lisk-elements/packages/cryptography.adoc[Cryptography module of Lisk Elements]   **Comment: please note this alternative does not seem to be described in this section, and both links are inactive - we need to check this.. (Chris - 13 Dec)

Reviewer: Yes, it is intended to only explain one alternative here. We just mention the second alternative for completion.
Concerning the links, have you checked on the live website? The links wont work on github, but should work when deployed to the website.

> Firstly, regarding the Lisk Commander, please ensure the Lisk Commander is installed in a secure environment.

Reviewer: ```suggestion
The first alternative with Lisk Commander is described in detail below.
Please ensure the Lisk Commander is installed in a secure environment.
```

      }'
----

* `publicKey` is the key for the delegate which is required to be enabled/disabled.
* `password` is the password used to encrypt your passphrase in `config.json`
* `forging` is the boolean value to enable or disable the forging.
* The HTTP Port can be different based on your configuration, therefore it is recommended to check the `httpPort` in your `config.json`

== SSL

[TIP]
====

> it is recommended to use a webserver such as https://www.nginx.com/[NGINX] or https://httpd.apache.org/[Apache] to set up the SSL for the Lisk Core.

Reviewer: ```suggestion
It is recommended to use a webserver such as https://www.nginx.com/[NGINX] or https://httpd.apache.org/[Apache] to set up the SSL for the Lisk Core.
```

If that opportunity is not available then it is also possible to configure the Lisk Core to handle SSL connections as described below:
====

[NOTE]
====
This step requires a signed certificate, (from a CA, such as https://letsencrypt.org[Let’s Encrypt]) or a self-signed certificate.

> Both the private and public keys will be needed in a location that is accessible to the Lisk core.

Reviewer: ```suggestion
Both the private and public keys will be needed in a location that is accessible to Lisk Core.
```

}
----

<1> Change from false to true.
<2> Default SSL Port.
<3> Change only if it is required to block web access to the node.

> <4> Replace from 'path_to_key' to the actual path to key file.

Reviewer: ```suggestion
<4> Replace `path_to_key` with the actual path to the key file.
```

> <5> Replace from 'path_to_cert' to the actual path to certificate file.

Reviewer: ```suggestion
<5> Replace `path_to_cert` with the actual path to the certificate file.
```

WARNING: If the SSL Port configured above in `http_api.ssl.options.port` is a privileged port, (below 1024), then either the node must be allowed to use the specified port with `setcap`, or the configuration must be changed in order to to use a port outside of that range.

*Setcap:* This is only required to grant Lisk access to port 443.

[source,bash]
----
 sudo setcap cap_net_bind_service=+ep bin/node
----

To verify your node has been configured correctly, open the web client using this url: `https://MY_IP_OR_HOST`.
A secure SSL connection should now be visible.

== Logging

> For monitoring or debugging your node, the Lisk Core tracks all activity that occurs within the node by creating applicable log messages.

Reviewer: I am unsure about adding "the" in front of Lisk Core, as it is the products name.
